### ISSUE TITLE
d_a_obj_Itnak

### ISSUE THREAD

**Issue by magcius** [zeldaret/tww#370]
Itnak

**Comment by ateddy42** [comment]
I'll try to tackle this one

**Comment by tommaso-perondi** [comment]
@ateddy42 i'm very sorry, i started decompiling like 10 minutes before your comment and forgot to comment myself.

**Comment by ateddy42** [comment]
Seems like yours is further along and this was still a good learning exercise, so I can close my PR. A few things I noticed on your branch:

- If you create a struct or class for L_attr it makes it a lot easier to reference the specific fields
- set_collision, _create, and _draw still have a lot of inline floats (68.0, 41.0, -120.0, 30.0, 800.0, etc.) that duplicate L_attr. Referencing L_attr fields would make this a lot easier to read.
- Looking at other d_a_obj* actors, I think you could add PRM_ enums instead of hardcoded values like 0 or 8.
- other reviews have called out variable names so they're a bit more understandable. ex: instead of `mInitParam` and `mParam`, I was calling those `mDrawMode` and `mVisible`, and variables like `bVar3` could be named `ret` or something more explicit about what it contains

**Comment by tommaso-perondi** [comment]
Good point, L_attr is actually already a struct (Attr_c) with named fields defined in the header, so it's really just a matter of wiring set_collision/_create/_draw to read from it instead of duplicating the same literals inline. 
I did try referencing the correct L_attr fields but couldn't get it to match. I'll take another look to see if I can fix it.
As for the variable names, I didn't want to use specific names if I wasn't sure of their use. I concur that mParam should be mVisible, but mInitParam seems to be the default value for mVisible, in my opinion it should be something like mInitVisible or mDefaultVisible.
And enums, yes, I should've added those earlier, I completely forgot.

**Comment by tommaso-perondi** [comment]
After some testing it seems like in the original code they defined L_attr but never used it. In the assembly you can see that for every value there is a separate lis+lfs instruction pair, each pointing to its own float literal in .rodata. When using L_attr the base address is calculated once and stored in a register. Then for every field the compiler loads(lfs) the values with their respective offset. Causing the object to not 100%match even if it's logically identical.

as for the rest i renamed all the variables to a more undertandable name and added the enums, thanks again for the suggestions.
